src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToRecord
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToRecord), Shared]
+    internal class CSharpConvertToRecordRefactoringProvider : CodeRefactoringProvider
+    {
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpConvertToRecordRefactoringProvider()
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, span, cancellationToken) = context;
+
+            var classDeclaration = await context.TryGetRelevantNodeAsync<TypeDeclarationSyntax>().ConfigureAwait(false);
+
+            if (classDeclaration == null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            if (semanticModel.GetDeclaredSymbol(classDeclaration, cancellationToken) is not INamedTypeSymbol type ||
+                // if type is some enum, interface, delegate, etc we don't want to refactor
+                (type.TypeKind != TypeKind.Class && type.TypeKind != TypeKind.Struct) ||
+                // TODO: when adding other options it might make sense to convert a postional record to a non-positional one and vice versa
+                // but for now we don't convert records
+                type.IsRecord ||
+                // records can't be static and so if the class is static we probably shouldn't convert it
+                type.IsStatic ||
```

Comment:
```suggestion
            if (semanticModel.GetDeclaredSymbol(classDeclaration, cancellationToken) is not INamedTypeSymbol
                {
                    // if type is some enum, interface, delegate, etc we don't want to refactor
                    TypeKind: TypeKind.Class or TypeKind.Struct,
                    // don't need to convert something that is already a record
                    IsRecord: false, 
                    // records can't be static and so if the class is static we probably shouldn't convert it
                    IsStatic: false,
                } type ||
                // TODO: when adding other options it might make sense to convert a postional record to a non-positional one and vice versa
```